src/App.tsx

```diff
@@ -15,33 +15,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer]
+
+  const [isVisible, setVisio] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [isSortByAlphabet, sortAlphabetically] = useState(false);
+  const [isSortByLength, sortByLength] = useState(false);
+  const [isReset, setResetedGoods] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goods.filter(good => good.length >= Number(goodsLength));
+
+  if (isSortByLength) {
+    visibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isSortByAlphabet) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+    }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  if (isReset) {
+    setReversed(false);
+    sortAlphabetically(false);
+    sortByLength(false);
+    setResetedGoods(false);
+    setGoodsLength('1');
+  }
+
+return (
+  <div className="App message">
+    {!isVisible && <div>
+      <p
+        className='title message-header'>
+          Control of goods starts from pushing this
+      </p>
+      <button
+        type="button"
+        className='button is-large is-fullwidth'
+        onClick={() =>setVisio(true)}
+      >
+        Start
     </button>
+    </div>
+    }
 
-    <button type="button">
+{isVisible && <div>
+    <button
+      type="button"
+      className='button is-dark'
+      onClick={() => {
+        sortAlphabetically(true);
+        sortByLength(false);
+        setReversed(false);
+      }}
+    >
       Sort alphabetically
     </button>
 
-    <button type="button">
+    <button
+      type="button"
+      className='button is-dark'
+      onClick={() => {
+        sortByLength(true);
+        sortAlphabetically(false);
+        setReversed(false);
+      }}
+    >
       Sort by length
     </button>
 
-    <button type="button">
+    <button
+      type="button"
+      className='button is-dark'
+      onClick={() => setReversed(!isReversed)}
+    >
       Reverse
     </button>
 
-    <button type="button">
+    <button
+      type="button"
+      className='button is-dark'
+      onClick={() => setResetedGoods(true)}
+    >
       Reset
     </button>
 
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
+    <ul className="Goods ">
+      {visibleGoods.map(good =>(
+        <li
+          className="Goods__item"
+          key={good}>{good}
+        </li>
+      ))}
     </ul>
+
+    <div>
+      <p className='message-body'>You can choose items to start length from value:</p>
+      <select name="select" value={goodsLength} onChange={({ target }) => setGoodsLength(target.value)}>
```

Comment:
what about formatting ?

Reply:
![image](https://user-images.githubusercontent.com/40194060/179574941-bf97868a-03dd-4db1-a52f-5b1f7f98fa57.png)


I don't understand, but on my screen it looks OK